appservice/src/deploy/runWithZipStream.ts

```diff
@@ -33,42 +35,29 @@ export async function runWithZipStream(context: IActionContext, fsPath: string,
         });
     } else {
         ext.outputChannel.appendLog(localize('zipCreate', 'Creating zip package...'), { resourceName: client.fullName });
+        let filesToZip: string[] = [];
 
-        // level 9 indicates best compression at the cost of slower zipping. Since sending the zip over the internet is usually the bottleneck, we want best compression.
-        const zipper: archiver.Archiver = archiver('zip', { zlib: { level: 9 } });
         if ((await fse.lstat(fsPath)).isDirectory()) {
             if (!fsPath.endsWith(path.sep)) {
                 fsPath += path.sep;
             }
 
+
             if (client.isFunctionApp) {
-                await addFilesGitignore(zipper, fsPath, '.funcignore');
+                filesToZip = await getFilesFromGitignore(fsPath, '.funcignore');
             } else {
-                addFilesGlob(zipper, fsPath);
+                filesToZip = await getFilesFromGlob(fsPath);
             }
         } else {
-            zipper.file(fsPath, { name: path.basename(fsPath) });
+            // remove the file from the fsPath
+            fsPath = fsPath.substring(0, fsPath.lastIndexOf(path.sep));
```

Comment:
What are you trying to do here? There has to be a `path` method that will do it for you

Reply:
I have to remove the file from the path name to pass into the zip function since it uses the deploy path and the file name as parameters.  I'll see if there's a path function that I can use, but this seemed like a simple enough action